### PR TITLE
Raise errors when theme can't be enabled

### PR DESCRIPTION
**Pull request by aleeks**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Raise errors when theme can't be enabled. Before page reload, theme wasn't enabled and we have no error |
| Type? | bug fix |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| How to test? | Install a theme with a missing template (form-fields.tpl) for example |

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

File: controllers/admin/AdminThemesController.php [hunk 1]
```diff
@@ -242,8 +242,13 @@ public function postProcess()
             $this->processSubmitConfigureLayouts();
             $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
         } elseif (Tools::getValue('action') == 'enableTheme') {
-            $this->theme_manager->enable(Tools::getValue('theme_name'));
-            $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
+            $themeHasEnabled = $this->theme_manager->enable(Tools::getValue('theme_name'));
```
Comment (kpodemski) [inline controllers/admin/AdminThemesController.php]: isThemeEnabled wouldn't be better?

**Review by mickaelandrieu — APPROVED** [review]

File: src/Core/Addon/Theme/ThemeValidator.php [hunk 2]
```diff
@@ -39,14 +47,24 @@ public function isValid(Theme $theme)
...
+
+                if (empty($translator)) {
+                    $translator = Context::getContext()->getTranslator();
+                }
+
+                if (!array_key_exists($themeName, $this->errors)) {
+                    $this->errors[$themeName] = array();
+                }
+
+                $this->errors[$themeName] = $translator->trans('An error occurred. The information "%s" is missing.', array($prop), 'Admin.Design.Notification');
```
Comment (mickaelandrieu) [inline src/Core/Addon/Theme/ThemeValidator.php]: ping the other Alex :p /c @AlexEven

Comment (aleeks) [inline src/Core/Addon/Theme/ThemeValidator.php]: It's ok for @AlexEven !

File: controllers/admin/AdminThemesController.php (hunk 1 above)
Comment (aleeks) [inline controllers/admin/AdminThemesController.php]: Done! Thanks @kpodemski

File: src/Core/Addon/Theme/ThemeValidator.php (hunk 2 above)
Comment (AlexEven) [inline src/Core/Addon/Theme/ThemeValidator.php]: Yes, seen with @aleeks :)
Thanks!

File: src/Core/Addon/Theme/ThemeManager.php [hunk 3]
```diff
@@ -211,6 +211,17 @@ public function getError($theme_name)
         return;
     }
 
+    /**
+     * Get all errors of theme install
+     *
+     * @param string $name The technical theme name
+     * @return array|false
+     */
+    public function getErrors($theme_name)
```
Comment (mickaelandrieu) [inline src/Core/Addon/Theme/ThemeManager.php]: $themeName

File: src/Core/Addon/Theme/ThemeManager.php [hunk 4]
```diff
@@ -211,6 +211,17 @@ public function getError($theme_name)
...
+    {
+        return $this->themeValidator->getErrors($theme_name);
```
Comment (mickaelandrieu) [inline src/Core/Addon/Theme/ThemeManager.php]: $themeName

File: src/Core/Addon/Theme/ThemeValidator.php [hunk 5]
```diff
@@ -27,9 +27,28 @@
...
+    private $translator;
+
+    private $errors = array();
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function getErrors($theme_name) {
```
Comment (mickaelandrieu) [inline src/Core/Addon/Theme/ThemeValidator.php]: $themeName

File: src/Core/Addon/Theme/ThemeValidator.php [hunk 6]
```diff
@@ -27,9 +27,28 @@
...
+        return array_key_exists($theme_name, $this->errors) ? $this->errors[$theme_name] : false;
```
Comment (mickaelandrieu) [inline src/Core/Addon/Theme/ThemeValidator.php]: $themeName

**Review by mickaelandrieu — CHANGES_REQUESTED** [review]

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Thank you @aleeks !

**Pull request merged**